program/src/state.rs

```diff
@@ -640,16 +628,15 @@ pub struct Validator {
     /// Sum of the balances of the stake accounts.
     pub stake_accounts_balance: Lamports,
 
-    /// Weight of the validator. Used when calculating the stake amount for
-    /// keeping a weighted balance, also defines the validator's share of fees.
-    pub weight: Weight,
+    /// Controls if a validator is allowed to have new stake deposits.
+    /// When removing a validator, this flag should be set to `true`.
+    pub inactive: bool,
 }
 
 impl Validator {
-    pub fn new(fee_address: Pubkey, weight: Weight) -> Validator {
+    pub fn new(fee_address: Pubkey) -> Validator {
         Validator {
             fee_address,
-            weight,
             ..Default::default()
```

Comment:
Let’s also initialize the field explicitly (maybe let’s just initialize all fields explicitly and remove the `Default`), I find it a bit scary to rely on an implicit default for such an important field.